### PR TITLE
ZenML Collaboration docs

### PR DESCRIPTION
**Pull request by stefannica**
## Describe changes
Add ZenML collaboration documentation section (preview here: https://app.gitbook.com/o/-MCl1Hlw9oU4xibZ-ymf/s/sSi8uXbe2nPYIyiHcpxS/collaborate/collaborate)

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/features/integrations) table.
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: docs/book/collaborate/collaborate.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+---
+description: Collaboration with ZenML.
+---
+
+# ZenML for Teams and Organizations
+
+Using ZenML to develop and execute pipelines from the comfortable confines of
+your workstation or laptop is a great way to incorporate MLOps best practices
+and production grade quality into your project from day one. However, as Machine
+Learning projects grow, managing and running ZenML in a single-user and
```
Comment (strickvl) [inline docs/book/collaborate/collaborate.md:10]: A general comment, but my preference would be for 'machine learning' and not 'Machine Learning'. WDYT?

File: docs/book/collaborate/collaborate.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+organizations.
+
+One such example is the decoupling of ZenML Stacks from Pipelines and their
+associated code, a principle that provides for a smooth transition from
+experimenting with and running Pipelines locally to deploying them to production
+environments. It is this same decoupling that also allows ZenML to be used
+in a setting where part of a team is iterating on writing ML code and
+implementing Pipelines, while the other part is defining and actively
+maintaining the infrastructure and the Stacks that will be used to execute
+Pipelines in production. Everyone can remain focused on their responsibilities
```
Comment (strickvl) [inline docs/book/collaborate/collaborate.md:31]: Same here (as elsewhere in this doc)... pipelines are common enough a term. Do we need to have it capitalised?

File: docs/book/collaborate/profiles.md [hunk 3]
```diff
@@ -0,0 +1,587 @@
...
+┃ ACTIVE │ STACK NAME │ ARTIFACT_STORE │ METADATA_STORE │ ORCHESTRATOR ┃
+┠────────┼────────────┼────────────────┼────────────────┼──────────────┨
+┃   👉   │ default    │ default        │ default        │ default      ┃
+┗━━━━━━━━┷━━━━━━━━━━━━┷━━━━━━━━━━━━━━━━┷━━━━━━━━━━━━━━━━┷━━━━━━━━━━━━━━┛
+```
+
+From the above example, you may have also noticed that _the active Profile_ and
+_the active Stack_ are global settings that affect all other user sessions open
+on the same machine. It is however possible to set the active Profile and active
+Stack individually for each user session or project. Keep reading to learn more.
```
Comment (strickvl) [inline docs/book/collaborate/profiles.md:164]: Cliffhanger!

File: docs/book/collaborate/profiles.md [hunk 4]
```diff
@@ -0,0 +1,587 @@
...
+store Profile data in a [local SQLite database file](#local-sqlite-profile),
+on [a remote MySQL server](#mysql-profile) or any SQL compatible database system
+for that matter. The URL value passed during the Profile creation controls the
+type, location and other parameters for the SQL database connection. To explore
+the full range of configuration options, consult the
+[SQLAlchemy documentation](https://docs.sqlalchemy.org/en/14/dialects/index.html).
+
+#### Local SQLite Profile
+
+The simplest form of SQL based Profile uses a SQLite file located in
```
Comment (strickvl) [inline docs/book/collaborate/profiles.md:419]:
```suggestion
The simplest form of SQL-based Profile uses a SQLite file located in
```

File: docs/book/collaborate/profiles.md [hunk 5]
```diff
@@ -0,0 +1,587 @@
...
+mysql -u zenml -p
+CREATE DATABASE zenml;
+```
+
+Then, on the client machine, some additional packages need to be installed (check
+[the SQLAlchemy documentation](https://docs.sqlalchemy.org/en/14/dialects/mysql.html)
+for the various MySQL drivers that are supported and how to use them):
+
+```
+sudo apt install libmysqlclient-dev
```
Comment (strickvl) [inline docs/book/collaborate/profiles.md:508]: Is it worth making a comment about how this applies to Debian Linux and other OSes must deal with it in their custom ways?

File: docs/book/collaborate/profiles.md [hunk 6]
```diff
@@ -0,0 +1,587 @@
...
+Finally, the command to create a new Profile would look like this:
+
+```
+$ zenml profile create --store-type sql --url "mysql://zenml:password@10.11.12.13/zenml" mysql_profile
+```
+
+## Migrating Stacks from Legacy Repositories to ZenML Profiles
+
+Traditionally, Stack configurations were stored locally in the Repository root
+directory - i.e. the local .zen folder that is created by the `zenml init`
```
Comment (strickvl) [inline docs/book/collaborate/profiles.md:521]:
```suggestion
directory - i.e. the local `.zen` folder that is created by the `zenml init`
```

File: docs/book/collaborate/profiles.md [hunk 7]
```diff
@@ -0,0 +1,587 @@
...
+command. The ZenML 0.7.0 release moves Stacks outside of Repository root
+folders, into the [global configuration directory](../developer-guide/repo-and-config.md).
```
Comment (strickvl) [inline docs/book/collaborate/profiles.md:523]:
```suggestion
folders into the [global configuration directory](../developer-guide/repo-and-config.md).
```

File: docs/book/collaborate/zenml-server.md [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+description: Use ZenML in a Collaborative Setting with the ZenServer.
+---
+
+# The ZenML Server
+
+Sometimes, you may need to exchange or collaborate on Stack configurations with
+other developers or even just have your Stacks available on multiple machines.
+While you can always zip up your Profile files or check [the local ZenStore](./profiles.md)
+files into version control, a more elegant solution is have some kind of service
+accessible from anywhere in your network and providing a REST API that can be
```
Comment (strickvl) [inline docs/book/collaborate/zenml-server.md:11]:
```suggestion
accessible from anywhere in your network that provides a REST API that can be
```

File: docs/book/collaborate/zenml-server.md [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+
+* [run a ZenServer locally](#running-the-zenserver-locally)
+* [deploy ZenServer with Docker](#deploy-the-zenserver-with-docker)
+* (more to come ...)
+
+
+{% hint style="info" %}
+The ZenServer is still undergoing heavy development. Some features are in Alpha
+state and may change in future releases. We are also working on providing more
+deployment and life-cycle management options for the ZenServer that will
```
Comment (strickvl) [inline docs/book/collaborate/zenml-server.md:34]:
```suggestion
deployment and lifecycle management options for the ZenServer that will
```

File: docs/book/collaborate/zenml-server.md [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+```
+$ zenml server up
+Running without an active repository root.
+Starting a new ZenServer local instance.
+ZenServer running at 'http://127.0.0.1:8000/'.
+```
+
+The local ZenServer exposes a local REST API through which clients can access
+the same data available in your active Profile. You can also specify a different
+profile by passing the `--profile=$PROFILE_NAME` command line argument, or change
```
Comment (strickvl) [inline docs/book/collaborate/zenml-server.md:82]:
```suggestion
profile by passing the `--profile=$PROFILE_NAME` command-line argument, or change
```

File: docs/book/collaborate/zenml-server.md [hunk 11]
```diff
@@ -0,0 +1,205 @@
...
+the HTTP port and address on which the ZenServer is accepting requests using
+`--port=$PORT` and `--ip-address=$IP_ADDRESS`.The following example creates
+a new SQL profile named `zen-server` and starts a ZenServer with that profile
+that listens to the 8080 port on all interfaces. This is a typical use case
```
Comment (strickvl) [inline docs/book/collaborate/zenml-server.md:86]:
```suggestion
that listens to port 8080 on all interfaces. This is a typical use case
```

**Review by strickvl — CHANGES_REQUESTED** [review]
Amazing new docs :) I learned a thing or two about the ZenServer! Some tiny nits / comments, but otherwise good to go from my side.

File: docs/book/collaborate/collaborate.md (hunk 1 above)
Comment (stefannica) [reply docs/book/collaborate/collaborate.md:10]: Your preference is good enough for me :) I'll change all occurrences.

File: docs/book/collaborate/collaborate.md (hunk 2 above)
Comment (stefannica) [reply docs/book/collaborate/collaborate.md:31]: agreed

File: docs/book/collaborate/profiles.md (hunk 5 above)
Comment (stefannica) [reply docs/book/collaborate/profiles.md:508]: Yes !

**Review by AlexejPenner — APPROVED** [review]
I just love this. Well done. The many code blocks make it very easy to follow along with the text around them. One small discussion item is the location of repo & config within our docs in general, but I'M happy merging this and dealing with the Table of contents at a later point.

:rocket:

File: docs/book/collaborate/collaborate.md (hunk 2 above)
Comment (strickvl) [reply docs/book/collaborate/collaborate.md]: My feeling was that Stefan was trying to denote ZenML-specific concepts with capitals. On the face of it, it's not a terrible idea. The only criterion in my mind is whether reading it as docs makes sense or not.

**Review by strickvl — APPROVED** [review]
🤓

**Review by htahir1 — APPROVED** [review]
Very well written!

**Pull request merged**